image/image.go

```diff
@@ -160,10 +172,13 @@ func sanitize(s string) string {
 	return strings.Replace(s, "/", "-", -1)
 }
 
+// compile-time check that manifestSchema2 implements genericManifest
+var _ genericManifest = (*manifestSchema2)(nil)
+
 type manifestSchema2 struct {
-	src    types.ImageSource
-	Config descriptor   `json:"config"`
-	Layers []descriptor `json:"layers"`
+	src               types.ImageSource
+	ConfigDescriptor  descriptor   `json:"config"`
+	LayersDescriptors []descriptor `json:"layers"`
```

Comment:
Might be worth naming these things `schema2Config` / `schema2Layers` (`s2…`?) to minimize pollution of the namespace — and they should definitely be non-public.